### PR TITLE
fix: set external module info

### PR DESCRIPTION
**Pull request by underfin**

### CODE REVIEW

File: crates/rolldown_common/src/types/module_info.rs [hunk 1]
```diff
@@ -12,3 +12,17 @@ pub struct ModuleInfo {
   pub imported_ids: Vec<ModuleId>,
   pub dynamically_imported_ids: Vec<ModuleId>,
 }
+
+impl From<ModuleId> for ModuleInfo {
```
Comment (hyf0) [inline crates/rolldown_common/src/types/module_info.rs]: I'll suggest using `ModuleInfo::new`.

**Review by hyf0 — APPROVED** [review]

File: crates/rolldown_common/src/types/module_info.rs (hunk 1 above)
Comment (underfin) [reply crates/rolldown_common/src/types/module_info.rs]: Here `ModuleInfo::new` needs to care about more fields, the external module only has `id`.

Comment (hyf0) [reply crates/rolldown_common/src/types/module_info.rs]: It's for clear reading

```diff
- self.shared_context.plugin_driver.set_module_info(&id.clone(), Arc::new(id.into()));
+ self.shared_context.plugin_driver.set_module_info(&id.clone(), Arc::new(moduleInfo::new(id)));
```

Comment (underfin) [reply crates/rolldown_common/src/types/module_info.rs]: > Here `ModuleInfo::new` needs to care about more fields, the external module only has `id`.

The `MouleInfo::new` does not follow the rust struct style, let using struct litreal at here.

**Pull request merged**